lib/money/money.rb

```diff
@@ -37,6 +37,19 @@ def from_subunits(subunits, currency_iso)
       new(value, currency)
     end
 
+    # Set Money.default_currency inside the supplied block, resets it to
+    # the previous value when done to prevent leaking state. Similar to
+    # ActiveSupport's Time.use_zone. This won't affect instances being created
+    # with explicitly set currency.
+    def use_currency(currency)
```

Comment:
what do you think of `with_currency` instead, I think it's closer to what's used elsewhere